x/accounts/internal/implementation/context.go

```diff
@@ -1,35 +1,98 @@
 package implementation
 
 import (
+	"bytes"
 	"context"
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
 
 	"cosmossdk.io/core/store"
 	"cosmossdk.io/x/accounts/internal/prefixstore"
 )
 
+var errUnauthorized = errors.New("unauthorized")
```

Comment:
Do we expect callers to be able to inspect and handle errors from handlers gracefully? If so, sentinel errors should be public.